apiserver/pkg/manager/resource_manager.go

```diff
@@ -10,3 +41,242 @@ func NewResourceManager(clientManager ClientManagerInterface) *ResourceManager {
+			configMap, err := r.GetComputeTemplate(ctx, name)
+			if err != nil {
+				return nil, err
+			}
+			computeTemplate := model.FromKubeToAPIComputeTemplate(configMap)
+			dict[name] = computeTemplate
+		}
+	}
+
+	return dict, nil
+}
+
+func (r *ResourceManager) GetCluster(ctx context.Context, clusterName string) (*v1alpha1.RayCluster, error) {
+	if len(clusterName) == 0 {
+		return nil, util.NewInvalidInputError("clusterName is empty, failed to get the cluster.")
+	}
+
+	client := r.getRayClusterClient(DefaultNamespace)
+	return getClusterByName(ctx, client, clusterName)
+}
+
+func (r *ResourceManager) ListClusters(ctx context.Context) ([]*v1alpha1.RayCluster, error) {
+	rayClusterList, err := r.getRayClusterClient(DefaultNamespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, util.Wrap(err, "List RayCluster failed")
+	}
+
+	var result []*v1alpha1.RayCluster
+	length := len(rayClusterList.Items)
+	for i := 0; i < length; i++ {
+		result = append(result, &rayClusterList.Items[i])
+	}
+
+	return result, nil
+}
+
+func (r *ResourceManager) DeleteCluster(ctx context.Context, clusterName string) error {
+	if len(clusterName) == 0 {
+		return util.NewInvalidInputError("clusterName is empty, failed to delete the cluster.")
+	}
+
+	client := r.getRayClusterClient(DefaultNamespace)
+	cluster, err := getClusterByName(ctx, client, clusterName)
+	if err != nil {
+		return util.Wrap(err, "Get cluster failure")
+	}
+
+	// Delete Kubernetes resources
+	if err := client.Delete(ctx, cluster.Name, metav1.DeleteOptions{}); err != nil {
+		// API won't need to delete the ray cluster CR
+		return util.NewInternalServerError(err, "Failed to delete cluster %v.", clusterName)
+	}
+
+	return nil
+}
+
+// Compute Runtimes
+func (r *ResourceManager) CreateComputeTemplate(ctx context.Context, runtime *api.ComputeTemplate) (*v1.ConfigMap, error) {
+	_, err := r.GetComputeTemplate(ctx, runtime.Name)
+	if err == nil {
+		return nil, util.NewAlreadyExistError("Compute template with name %s already exists in namespace %s", runtime.Name, DefaultNamespace)
+	}
+
+	computeTemplate, err := util.NewComputeTemplate(runtime, DefaultNamespace)
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to convert compute runtime (%s/%s)", DefaultNamespace, runtime.Name)
+	}
+
+	client := r.clientManager.KubernetesClient().ConfigMapClient(DefaultNamespace)
+	newRuntime, err := client.Create(ctx, computeTemplate, metav1.CreateOptions{})
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to create a compute runtime for (%s/%s)", DefaultNamespace, runtime.Name)
+	}
+
+	return newRuntime, nil
+}
+
+func (r *ResourceManager) GetComputeTemplate(ctx context.Context, name string) (*v1.ConfigMap, error) {
+	client := r.clientManager.KubernetesClient().ConfigMapClient(DefaultNamespace)
+	return getComputeTemplateByName(ctx, client, name)
+}
+
+func (r *ResourceManager) ListComputeTemplates(ctx context.Context) ([]*v1.ConfigMap, error) {
```

Comment:
Is computeTemplate saved in configmap?

Reply:
Yes. That's true. We can change to db or annotations if needed later.